src/bindgen/processor/runner.cr

```diff
@@ -8,7 +8,7 @@ module Bindgen
 
       def initialize(config : Configuration, db : TypeDatabase)
         @processors = config.processors.map do |name|
-          Processor.create_by_name(Processor::ERROR_KIND, name, config, db)
+          Processor.create_by_name(Processor::ERROR_KIND, name, config, db).as(Processor::Base)
```

Comment:
`Generator` and `Processor.create_by_name` are generated by the generic module `CreateByName`.  You should only need to do the same as for Checker, and add the returning type to `.create_by_name`.  (It should explicitly return  `T`)